spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -136,7 +136,7 @@ public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceExcepti
   public Table loadTable(Identifier ident) throws NoSuchTableException {
     try {
       return icebergCatalog.loadTable(ident);
-    } catch (NoSuchTableException e) {
+    } catch (NoSuchTableException | org.apache.iceberg.exceptions.NotFoundException e) {
```

Comment:
Why do we only do this in SparkSessionCatalog? Shouldn't the same issue exist with SparkCatalog? I also wonder if we should have a session parameter or something to enable this behavior. Basically in this case we are saying drop table should drop any table that has a catalog entry even if we can't determine it is an Iceberg table.

I think we can probably make this a little safer by checking whether the table has the metadata_location property set in the catalog? I'm open to other suggestions as well.